examples/cache.ex

```diff
@@ -1,74 +1,125 @@
+  # The catch all event case that takes and ignores all events you are not
+  # interested in, if you do not have this, or have not defined literally
+  # every event case yourself (what an absolute madlad if you have),
+  # the consumer will crash, not having any function signature to match upon.
+  # By the way, the return atom stands for "no-op", shorthand for "no operation",
+  # however, if you read it as "noop" and quietly chuckle to yourself every time you see it,
+  # since it just sounds like a silly way of saying "nope", know, that you are not alone.
+  def handle_event(_event), do: :noop
+end
 
-        # They typed an invalid input, probably due to using letters rather than numbers.
-        :error ->
-          Api.create_message!(msg.channel_id, "Make sure the User ID is only numbers")
+# Our basic, example command handler that will be taking the message
+# and actually doing something with it, wow, amazing!
+defmodule ExampleCommands do
+  import Nostrum.Snowflake, only: [is_snowflake: 1]
 
-        # There wasn't 2 elements in there, so it returned false.
-        false ->
-          Api.create_message!(msg.channel_id, "Please supply a User ID")
-      end
-    end
+  alias Nostrum.Api
+  alias Nostrum.Cache.{ChannelCache, GuildCache, UserCache}
+  alias Nostrum.Struct.User
 
-    {:ok, state}
+  # Fetch the defined prefix from our config, however,
+  # if you have not defined it in your config yet, fear not,
+  # the line below will default it to "!"
+  @prefix Application.get_env(:example_app, :prefix, "!")
+
+  defp get_cached_with_fallback(id, cache, api) do
+    case cache.(id) do
+      {:ok, _} = data -> data
+      {:error, _} -> api.(id)
+    end
   end
 
-  # Default event handler, if you don't include this, your consumer WILL crash if
-  # you don't have a method definition for each event type.
-  def handle_event(_, state) do
-    {:ok, state}
+  # The command that will be handling our message.
+  # Pattern matching's pretty neat by the way, a nice way to define a shape for the data we accept
+  # and have it reject the rest, also gets rid of a good chunk of all that pesky condition checking
+  # that you would normally see, trying for that data shape that we need, destructuring's also pretty neat.
+  # For this, a simple `userinfo` command that accepts a user id and returns a message
+  # containing some things you would need to fetch from the cache (or the API, as a backup).
+  def command(%{
+        content: @prefix <> "userinfo " <> message_user_id,
+        channel_id: channel_id,
+        guild_id: guild_id
+      })
+      when guild_id != nil do
+    # Our happy path for sending a complete userinfo response.
+    # Pretty standard stuff, maybe apart from additionally wrapping both sides in a tuple
+    # with an extra identifier, that's just to make failure cases more obvious.
+    with {{user_id, _binary}, :parse_id} when is_snowflake(user_id) <-
```

Comment:
I don't think this is a bad example, but given this will probably be used by beginners mostly (I think?) something a little more simple might be better. I think seeing all the curly braces might discourage some people.

Reply:
Maybe we just need to get rid of the resource portion?